components/base/Button/Button.tsx

```diff
@@ -0,0 +1,41 @@
+type TButtonSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl';
+type TButtonVariation = 'default' | 'outlined';
+
+const buttonSizes: Record<TButtonSize, string> = {
+  sm: 'text-sm px-[14px] py-2',
+  md: 'text-sm px-4 py-[10px]',
+  lg: 'px-[18px] py-[10px]',
+  xl: 'px-5 py-3',
+  '2xl': 'text-lg px-7 py-4'
+};
+
+const buttonBase = 'font-semibold rounded-lg shadow-sm shadow-primary-100/5';
```

Comment:
Apparently `cva` don't support that behavior. They recommended a wrapper element in their document. 
Also, additional class names can be added to component while using.

```tsx
<Button size="md" variant="outline" className="md:px-14 xl:px-20">...
```

If you think it is mandatory, I will try to find a way. How would API be in that case?

Reply:
yes, it is mandatory but I did not understand it well enough. what is not supporting? maybe we can create a custom object that returns a specific size variant? What do you think?

Reply:
At the last part of their documentation: *"cva doesn't know about how you choose to apply CSS clases, and it doesn't want to."*.

```ts

Reply:
It's in the last part of cva readme.
> "cva doesn't know about how you choose to apply CSS clases, and it doesn't want to.".

---
I can achieve this by adding additional property like `sizeAfterMobile` *-need better naming for sure-*.  Which will add `md:text-sm md:px-4` utility classes to the button and css will handle the rests.

It means literally duplicating `size` object below and adding `md:` or `lg:` to each string.
https://github.com/frontendship/frontendship-website/blob/1087687040c11dced592b34ef6b701bee8d04553/components/base/Button/Button.tsx#L33-L38

And the usage would be:
```ts
<Button size="md"> // default stays same
<Button size="md" sizeAfterMobile="lg" sizeAfterTablet="xl"> // responsive
```

Reply:
It looks messy. Maybe we can create an object that includes multiple breakpoint values. And we can find the matched variant from the CVA variants? what do you think?

Reply:
Problem is, `cva` just returning some strings after function call, to deal with breakpoints with `CSS` we need to modify those strings afterwards. I guess creating `sizeAfterMobile` internally would be better than recursively appending tailwind classes with `md:` etc.

```ts
<Button size="md"> // valid, no media query
<Button size=["md", "lg", "xl"]> // valid with media queries
```

If you want this kind of API, I guess we need to create JS logic to decide if it is array with 3 elements or string or undefined. 
API will be simpler but I don't know how to deal with it without messy way inside the component 😥.  If there is an example I would love to implement that though. 

IMHO it felts like it is beyond the scope of button component.

Few ideas:

- We can create another dynamic `Box` kind of component to deal with that concern and pass `<Button fullWidth>` to it.
- We can limit this to only `fullWidth`
```ts
<Button fullWidth> // full width in every case
<Button fullWidth="md"> // full width after md breakpoint
```

Reply:
It can be complicated if we give breakpoint values to full Width. We can manage this responsive problem by containing additional wrapper at pages which button component used as you mention. I think we can merge this with this structure.